runtimes/runtimes/chat/encryptedChat.ts

```diff
@@ -0,0 +1,156 @@
+import { jwtDecrypt, EncryptJWT, JWTPayload } from 'jose'
+import { Connection } from 'vscode-languageserver'
+import {
+    ChatParams,
+    ChatResult,
+    EndChatParams,
+    FeedbackParams,
+    FollowUpClickParams,
+    InfoLinkClickParams,
+    InsertToCursorPositionParams,
+    LinkClickParams,
+    NotificationHandler,
+    QuickActionParams,
+    QuickActionResult,
+    RequestHandler,
+    SourceLinkClickParams,
+    TabAddParams,
+    TabChangeParams,
+    TabRemoveParams,
+    chatRequestType,
+    feedbackNotificationType,
+    readyNotificationType,
+    tabAddNotificationType,
+    tabChangeNotificationType,
+    tabRemoveNotificationType,
+    insertToCursorPositionNotificationType,
+    linkClickNotificationType,
+    infoLinkClickNotificationType,
+    sourceLinkClickNotificationType,
+    followUpClickNotificationType,
+    endChatRequestType,
+    CancellationToken,
+    EncryptedChatParams,
+    EncryptedQuickActionParams,
+    quickActionRequestType,
+} from '../../protocol'
+import { Chat } from '../../server-interface'
+import { CredentialsEncoding } from '../auth/standalone/encryption'
+
+export class EncryptedChat implements Chat {
```

Comment:
Most of methods implemented here are duplicating non-encrypted version of Chat in the runtime. It's better to convert non-encrypted object to new class and move duplicated methods to shared parent abstract class. This way encryption-specific implementations classes will be much leaner

Reply:
Do we even need strong typing of the encrypted versions? Can we operate encryption at the lower level `params` and `results` members of the `json-rpc` requests/responses/notifications?